k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java

```diff
@@ -27,18 +27,16 @@
 
 
 public class LocalMessage extends MimeMessage {
-    protected MessageReference mReference;
     private final LocalStore localStore;
 
-    private long mId;
-    private int mAttachmentCount;
-    private String mSubject;
-
-    private String mPreview = "";
-
-    private long mThreadId;
-    private long mRootId;
+    private long databaseId;
```

Comment:
I think I disagree here.

Messages have various of ids: a database id, a messageId, a uid, a message part id, a thread id, a thread root id, and this in a class that has a complex superclass. Especially "id" and "message id" are easy to mix up, and in fact we managed to mix up uid and database id in the codebase (see 2fbd6f61a886a822f6953ab8cc374c8e0ca2db59). We should make sure this doesn't happen again.

Also consider spoken language: I'm pretty sure I would converge towards calling this id the "database id" sooner or later, to make sure the other person knows which one I'm talking about.